docs/tools/config_system.md

```diff
@@ -34,28 +147,40 @@ For example:
+  - `macro_name`: an optional name for the macro defined at compile time for this configuration parameter. The configuration system will automatically figure out the corresponding macro name for a configuration parameter, but the user can override this automatically computed name by specifying `macro_name`.
 
-- *The short way:* by name and value. `param3` above is an example of a short definition for a parameter named `param3` with value `10`.
-- *The long way:* by name and description (another JSON object), like `param1` and `param2` above. The JSON description object can have the following keys:
-    - `help`: an optional help message that describes the purpose of the parameter.
-    - `value`: an optional field that defines the value of the parameter.
-    - `required`: an optional key that specifies whether the parameter **must** be given a value before compiling the code (`false` by default). It's not possible to compile a source tree with one or more required parameters that don't have a value. Generally, it makes sense to define a required parameter only when it doesn't have a `value` key.
-    - `macro_name`: an optional name for the macro defined at compile time for this configuration parameter. The configuration system will automatically figure out the corresponding macro name for a configuration parameter, but the user can override this automatically computed name by specifying `macro_name`.
+You define a macro by value by using an integer or string instead of the deception object, such as `param3` above. Defining a parameter by value is equivalent to a configuration parameter defined with a description object with the key `help` unset, the key `macro_name` unset, the key `required` set to `false`, and the key `value` set to the value in place of the description object. 
 
 <span class="notes">**Note:** the name of a parameter in `config` can't contain a dot (`.`) character.</span>
 
-The configuration system automatically appends an *implicit prefix* to the name of each parameter, so you don't have to worry about a name clash if you define a parameter with the same name in a library and a target, for example. The implicit prefix is:
+The configuration system appends a prefix to the name of each parameter, so a parameter with the same name in a library does not conflict with parameters of the same name in targets or other libraries. The prefix is:
 
-- **target.** if the parameter is defined in a target.
-- **app.** if the parameter is defined in the application.
-- The name of the library followed by a dot (.) if the parameter is defined in a library.
+| Location | Prefix |
+| -------- | ------ |
+| target | `target.` |
+| any library | The name of the library, as found in the `name` section of `mbed_lib.json`, followed by a dot (.) |
```

Comment:
Since my current belief is that mbed_lib.json is optional.
What if the library doesn't use mbed_lib.json?

Reply:
Then it can't define any parameters, and the prefix may remain undefined.

Reply:
Same goes for the test framework?

Reply:
^ ignore meant to comment on another conversation.

Reply:
I'm not sure what you mean.

If you're refering to utest, It's `mbed_lib.json` is below
```JSON
{                                                                                                        
    "name": "utest",                                                                                     
    "macros": ["UNITY_INCLUDE_CONFIG_H"]                                                                 
} 
```